src/include/txn_inline.h

```diff
@@ -224,6 +225,19 @@ __txn_next_op(WT_SESSION_IMPL *session, WT_TXN_OP **opp)
     op = &txn->mod[txn->mod_count++];
     WT_CLEAR(*op);
     op->btree = S2BT(session);
+
+    /*
+     * Store the ID of the latest transaction that is making an update. It can be used to determine
+     * if there is an active transaction on the btree. Retry if we race with another thread, but
+     * give up if that update is newer than this one.
+     */
+    btree_txn_id_prev = op->btree->max_upd_txn;
```

Comment:
In reference to the corresponding check in ```__wt_conn_dhandle_close()```, I think we should save the latest txn-id only if the change comes from an application thread. We can do something like:

```
if (!F_ISSET(session, WT_SESSION_INTERNAL)) {
    /* 
     * Store the ID ...
     *
     */
    btree_txn_id_prev = op->btree->max_upd_txn;
    while (WT_TXNID_LT(btree_txn_id_prev, txn->id)) {
        if (__wt_atomic_cas64(&op->btree->max_upd_txn, btree_txn_id_prev, txn->id))
            break;
        btree_txn_id_prev = op->btree->max_upd_txn;
    }
}
```

Then I am guessing that the check in ```__conn_dhandle_close_one()``` will not have to skip checkpoint handles and sweep sessions.

Reply:
This did not work for sweep sessions.

Reply:
Since this did not work, To make it easier to see what is left I am going to resolve some conversations.